### PR TITLE
Feature/ele 204 client side api structure for using the opensearch index

### PR DESCRIPTION
**Pull request by dannelundqvist**
Beginning of a structure to search against the index using Planning. Renamed it to PlanningOverview to not collide with Planning api object. Refactored session and jwt handling a bit as well to accomodate using the session provider for more than the jwt in the future.

### CODE REVIEW

File: src/views/Editor/index.tsx [hunk 1]
```diff
@@ -35,7 +35,7 @@ export const Editor = (props: ViewProps): JSX.Element => {
     return new HocuspocusProvider({
       websocketProvider: hocuspocusWebsocket,
       name: documentId,
-      token: jwt.accessToken as string,
+      token: jwt.access_token as string,
```
Comment (gularsson) [inline src/views/Editor/index.tsx:38]: `access_token` falls under `JWTPayload`s catch all typing `[propName: string]: unknown`. Should probably be added to our `JWT` type, then we dont have to cast it as string.

**Review by gularsson — CHANGES_REQUESTED** [review]
Awesome, great work. Session/jwt handling looks like I imagined after our talk, and searching looks smooth. Some minor changes maybe, some suggestions and some questions below.

File: .env.tpl [hunk 2]
```diff
@@ -13,6 +13,9 @@ JWKS_URL="https://.../.well-known/jwks.json"
 REDIS_HOST="localhost"
 REDIS_PORT=6379
 
+# Opensearch index
+VITE_INDEX_URL="https://...se:443"
```
[the remark this answers is not in the document]
Comment (dannelundqvist) [reply .env.tpl:17]: Your'e probably right. It's only an example in .env template though.

File: src/lib/planning/search.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  )
+}
+
+const query = {
+  query: {
+    bool: {
+      must: [
+        {
+          term: {
+            'document.meta.core_assignment.data.start_date': '2023-10-01T00:00:00Z'
```
[the remark this answers is not in the document]
Comment (dannelundqvist) [reply src/lib/planning/search.ts:21]: Oops. I'll fix.

File: src/views/Editor/index.tsx (hunk 1 above)
Comment (dannelundqvist) [reply src/views/Editor/index.tsx:38]: True!

File: .env.tpl (hunk 2 above)
[the remark this answers is not in the document]
Comment (gularsson) [reply .env.tpl:17]: true :+1:

File: src/lib/planning/search.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (dannelundqvist) [reply src/lib/planning/search.ts]: I moved it to the calling view, still hardcoded. We can fix that when implementing the actual functionality of the view.

File: src/main.tsx [hunk 4]
```diff
@@ -8,10 +8,11 @@ const host = import.meta.env.VITE_API_HOST
 const port = import.meta.env.VITE_API_PORT
 const apiUrl = `${protocol}://${host}:${port}/api`
 const websocketUrl = `ws://${host}:${port}/ws`
+const indexUrl = import.meta.env.VITE_INDEX_URL
 
 ReactDOM.createRoot(document.getElementById('root') as HTMLElement).render(
   <SessionProvider endpoint={`${apiUrl}/user`}>
-    <ApiProvider apiUrl={new URL(apiUrl)} websocketUrl={new URL(websocketUrl)}>
+    <ApiProvider apiUrl={new URL(apiUrl)} websocketUrl={new URL(websocketUrl)} indexUrl={new URL(indexUrl)}>
```
[the remark this answers is not in the document]
Comment (dannelundqvist) [reply src/main.tsx:15]: Yes!

**Review by gularsson — APPROVED** [review]

**Pull request merged**